cvat-ui/src/reducers/selection-reducer.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import { SelectionState } from 'reducers';
+import {
+    SelectionActionsTypes,
+    SelectionActions,
+} from '../actions/selection-actions';
+
+const initialState: SelectionState = {
+    selected: [],
+    fetching: false,
+    status: null,
+    cancelled: false,
+};
+
+export default function selectionReducer(
+    state = initialState,
+    action: SelectionActions,
+): SelectionState {
+    switch (action.type) {
+        case SelectionActionsTypes.DESELECT_RESOURCES:
+            return { ...state, selected: state.selected.filter((id) => !action.payload.resourceIds.includes(id)) };
+        case SelectionActionsTypes.SELECT_RESOURCES:
+            if (action.payload.extendSelection) {
+                return { ...state, selected: Array.from(new Set([...state.selected, ...action.payload.resourceIds])) };
+            }
+            return { ...state, selected: [...action.payload.resourceIds] };
+        case SelectionActionsTypes.CLEAR_SELECTED_RESOURCES:
+            return { ...state, selected: [] };
```

Comment:
Seems like redundant action. You may just call SELECT_RESOURCES with empty array.

Reply:
I’d prefer to keep this action. The current `Select` action works as **“append some resources”**, not **“fully replace selected resources.”**

This is useful when we want to append items (e.g., via Ctrl or Shift click), so we don’t need to worry about some previously selected ones. It keeps the "set" logic centralized at the reducer level.